### ISSUE TITLE
ExceptionHandlerExceptionResolver.setResponseBodyAdvice() Javadoc is incorrect

### ISSUE THREAD

**Issue by sbrannen** [spring-projects/spring-framework#24015]
The Javadoc for `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()` states that it will "Add one or more components..."; however, it actually first clears the current list and effectively _sets_ the advice to the specified list.

This is in contrast to `RequestMappingHandlerAdapter.setResponseBodyAdvice()` which actually does _add_ the advice instead of clearing the current list.

Granted the name of the latter is misleading and should actually be `addResponseBodyAdvice()`, we likely will not want to change the semantics.

Thus, we should consider one of the following.

1. Update Javadoc for `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()` to reflect that it actually "sets" the list instead of adding.
2. Stop invoking `this.responseBodyAdvice.clear()` so that `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()` does what the Javadoc says it does.

Option # 2 would align the behavior in `ExceptionHandlerExceptionResolver` and `RequestMappingHandlerAdapter`.

**Comment by sbrannen** [comment]
**Team Decision**: go with option # 2 but not until Spring Framework 5.3.

> 2. Stop invoking `this.responseBodyAdvice.clear()` so that `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()` does what the Javadoc says it does.

**Comment by ahmad222223** [comment]
Ok mr

**Comment by ahmad222223** [comment]
> تنص Javadoc `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()`على أنه " سيضيف مكونًا واحدًا أو أكثر ..."؛ ومع ذلك ، فإنه أولاً أولاً مسح القائمة الحالية _وتعيين_ المشورة إلى القائمة المحددة بشكل فعال .
> 
> هذا على النقيض `RequestMappingHandlerAdapter.setResponseBodyAdvice()`مما _يضيف_ النصائح بالفعل بدلاً من مسح القائمة الحالية.
> 
> منحت اسم الأخير مضللاً ويجب أن يكون في الواقع `addResponseBodyAdvice()`، من المحتمل أننا لن نريد تغيير الدلالات.
> 
> وبالتالي ، يجب علينا النظر في واحد مما يلي.
> 
> 1. قم بتحديث Javadoc `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()`ليعكس أنه في الواقع "يحدد" القائمة بدلاً من الإضافة.
> 2. توقف عن الاستدعاء `this.responseBodyAdvice.clear()`بحيث `ExceptionHandlerExceptionResolver.setResponseBodyAdvice()`يفعل ما يقوله Javadoc.
> 
> الخيار رقم 2 من شأنه محاذاة السلوك في `ExceptionHandlerExceptionResolver`و `RequestMappingHandlerAdapter`.

**Comment by fahadal213344** [comment]
ازااللبب

**Comment by rstoyanchev** [comment]
@ahmad222223, I've deleted your non-English comments. If did not leave those in error, and actually meant to comment, perhaps try Google translate or similar.

**Comment by rohitp27** [comment]
Hey I'd like to start contributing to spring. I could take up the documentation change. If this isn't possible then could someone more experienced guide me on a smaller issue to get started with? 

Thanks

**Comment by rstoyanchev** [comment]
This issue is small enough if you'd like to give that a try. We also use the `status: ideal-for-contribution` label for issues that are straight-forward enough.

**Comment by rohitp27** [comment]
Great! I'll take this up then.

**Comment by rohitp27** [comment]
@rstoyanchev I've created a PR. Could you get it approved?

**Comment by sbrannen** [comment]
Superseded by #24204.
